src/org/labkey/targetedms/TargetedMSController.java

```diff
@@ -1464,16 +1464,30 @@ private GuideSet getClosestPastGuideSet(List<GuideSet> guideSets, Date startDate
             for (GuideSet guideSet : guideSets)
             {
                 // guideset end date should be before or equal start date
-                if (!guideSet.isDefault() &&
-                        (guideSet.getTrainingEnd().before(startDate) || DateUtil.getDateOnly(guideSet.getTrainingEnd()).compareTo(startDate) == 0))
+                if (!guideSet.isDefault())
                 {
-                    if (null == gs)
+                    // show reference guideset when
+                    // 1. startDate is after guideset
+                    // 2. startDate is between guideset date range
+                    // 3. startDate is same as guideset training start
+                    // 4. startDate is same as guideset training end
+                    boolean startDateAfterGuideSet = guideSet.getTrainingEnd().before(startDate);
+                    boolean startDateBetweenGuideSet = guideSet.getTrainingStart().before(startDate) && guideSet.getTrainingEnd().after(startDate);
+                    boolean startDateOverlapGuideSetStart = DateUtil.getDateOnly(guideSet.getTrainingStart()).compareTo(DateUtil.getDateOnly(startDate)) == 0;
```

Comment:
`compareTo() == 0` is effectively `equals()`, isn't it?

Reply:
yes